lib/pipefy_message/consumer.rb

```diff
@@ -47,33 +48,15 @@ module ClassMethods
       # Merges default worker options with the hash passed as
       # an argument. The latter takes precedence.
       def options(opts = {})
-        @options = Consumer.default_consumer_options.merge(opts)
-        @options.each do |k, v|
-          singleton_class.class_eval { attr_accessor k }
-          send("#{k}=", v)
-        end
-
-        logger.debug({
-                       options_set: @options,
-                       message_text: "Set #{name} options to options_set"
-                     })
-      end
-
-      ##
-      # Sets broker-specific options to be passed to the broker's
-      # constructor.
-      def broker_options(opts = {})
-        @broker_opts = opts
+        @consumer_options = Consumer.default_consumer_options.merge(opts)
       end
 
       ##
       # Initializes and returns an instance of a broker for
       # the provider specified in the class options.
       def build_consumer_instance
-        options if @broker.nil?
-        broker_options if @broker_opts.nil?
-        consumer_map = resolve_broker(@broker, "consumer")
-        consumer_map[:class_name].constantize.new(@broker_opts)
+        consumer_map = resolve_broker(@consumer_options[:broker], "consumer")
+        consumer_map[:class_name].constantize.new(@consumer_options)
```

Comment:
**suggestion (non-blocking)**: I guess that's not currently an issue, but maybe we could exclude the "broker" key and its value from the hash that gets passed on to the consumer constructor, just so we don't run into any issues with extraneous options being passed. But that's not urgent, I just wanted to highlight that this is currently happening.

Reply:
When we have more than one provider we'll have this situation too, so maybe we can add more validation or change the parameter type, but not now... I suggest to keep as is and move forward.

Also, the broker resolver will fail in case of extraneous options, so that’s not a big problem  😄